src/game/common/system/mempool.cpp

```diff
@@ -180,6 +182,8 @@ int MemoryPool::Release_Empties()
 
 void MemoryPool::Reset()
 {
+    ScopedCriticalSectionClass scs(g_memoryPoolCriticalSection);
```

Comment:
Are all locks now placed in this class? Asking because the lock coverage looks wild...